modules/renter/hostdb/hostdb.go

```diff
@@ -75,9 +75,9 @@ type hostContract struct {
 	SecretKey       crypto.SecretKey
 }
 
-// New creates and starts up a hostdb. The hostdb that gets returned will not
-// have finished scanning the network or blockchain.
-func New(cs modules.ConsensusSet, wallet modules.Wallet, tpool modules.TransactionPool, persistDir string) (*HostDB, error) {
+// New returns a new HostDB.
+func New(cs consensusSet, wallet walletShim, tpool transactionPool, persistDir string) (*HostDB, error) {
```

Comment:
I don't think `walletShim` is the right name for the dependency. What you need is a wallet, not a shim. If you want to use the name shim, it should probably be on the other interface. It just seems weird to me that a dependency would be a shim.